packages/asset-swapper/contracts/src/BalanceChecker.sol

```diff
@@ -0,0 +1,136 @@
+/*
+
+  Copyright 2020 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.5.9;
+
+// ERC20 contract interface
+contract IToken {
+    /// @dev Query the balance of owner
+    /// @param _owner The address from which the balance will be retrieved
+    /// @return Balance of owner
+    function balanceOf(address _owner) public view returns (uint);
+
+    /// @param _owner The address of the account owning tokens
+    /// @param _spender The address of the account able to transfer the tokens
+    /// @return Amount of remaining tokens allowed to spent
+    function allowance(address _owner, address _spender) public view returns (uint);
+}
+
+contract BalanceChecker {
+    /* Fallback function, don't accept any ETH */
+    function() external payable {
+      revert("BalanceChecker does not accept payments");
+    }
+
+    /*
+      Check the token balance of a wallet in a token contract
+
+      Returns the balance of the token for user. Avoids possible errors:
+        - return 0 on non-contract address 
+        - returns 0 if the contract doesn't implement balanceOf
+    */
+    function tokenBalance(address user, address token) public view returns (uint) {
```

Comment:
In our use case, wouldn't it be better for the contract to simply revert in the case a token address does not exist, instead of failing silently? do you think that this error could ever happen outside of a bug that we could have introduced =?

Reply:
👍  change to revert

Reply:
Nit: `uint->uint256` errwhere